lib/bike_brigade_web/live/campaign_live/duplicate_campaign_component.ex

```diff
@@ -57,26 +82,40 @@ defmodule BikeBrigadeWeb.CampaignLive.DuplicateCampaignComponent do
   end
 
   @impl Phoenix.LiveComponent
-  def handle_event("duplicate", %{"duplicate_form" => duplicate_form}, socket) do
+  def handle_event("duplicate", %{"duplicate_form" => duplicate_form_params}, socket) do
+    %{
+      "date_time_form" => date_time_form_params,
+      "duplicate_deliveries" => duplicate_deliveries,
+      "duplicate_riders" => duplicate_riders
+    } = duplicate_form_params
+
     old_campaign = socket.assigns.campaign
 
-    {:ok, new_campaign} = create_new_campaign(old_campaign, duplicate_form)
+    {:ok, new_campaign} = create_new_campaign(old_campaign, date_time_form_params)
 
     if old_campaign.instructions_template_id do
       copy_instructions_template(old_campaign, new_campaign)
     end
 
+    if duplicate_deliveries == "true" do
+      copy_delivery_tasks(old_campaign, new_campaign)
```

Comment:
Probably okay to leave as a follow up but if there are a lot of tasks/riders this could be really slow & time out. Is there a phoenix/elixer-y way of handling this in a background job?